nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/controller/FlowController.java

```diff
@@ -1365,6 +1376,97 @@ private ProvenanceRepository createProvenanceRepository(final NiFiProperties pro
         }
     }
 
+    private NarPersistenceProvider createNarPersistenceProvider(final NiFiProperties nifiProperties) {
```

Comment:
Is there a reason for creating the implementation in this class? It seems cleaner to define it in the Spring configuration, and then pass the `NarPersistenceProvider` to the constructor, or through a setter.

Reply:
Good point, it really doens't need to be in FlowController. I will extract the creation to a separate factory bean that can be called from the new Java Spring config.